packages/legend-graph/src/models/protocols/pure/v1/engine/V1_Engine.ts

```diff
@@ -471,12 +471,16 @@ export class V1_Engine {
     generationMode: GenerationMode,
     model: V1_PureModelContextData,
   ): Promise<V1_GenerationOutput[]> {
+    const grammar = this.pureModelContextDataToPureCode(model);
```

Comment:
Please add a `TODO` here for us to revisit and try to use `PureModelContextText`. This call looks really weird and potentially super expensive for other generation flow in terms of network traffic as it includes source information!!

Reply:
Added ToDo for revisiting.